lib/src/encoder/compact_array.dart

```diff
@@ -0,0 +1,36 @@
+part of 'encoder.dart';
+
+class _CompactArray<T> extends ByteArray {
```

Comment:
It doesn't make sense for this class to be generic, it doesn't use this parameter inside.

Reply:
True. It did use it initially. Well caught :)